agent/checks/check_test.go

```diff
@@ -536,6 +536,72 @@ func TestCheckHTTPTimeout(t *testing.T) {
 	})
 }
 
+func TestCheckHTTPBody(t *testing.T) {
+	t.Parallel()
+	timeout := 5 * time.Millisecond
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		code := 200
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			code = 999
+			body = []byte(err.Error())
+		}
+
+		w.WriteHeader(code)
+		w.Write(body)
+	}))
+	defer server.Close()
+
+	tests := []struct {
+		desc   string
+		method string
+		header http.Header
+		body   string
+	}{
+		{desc: "get body", method: "GET", body: "hello world"},
+		{desc: "post body", method: "POST", body: "hello world"},
+		{desc: "post json body", header: http.Header{"Content-Type": []string{"application/json"}}, method: "POST", body: "{\"foo\":\"bar\"}"},
+	}
+
+	for _, tt := range tests {
+		desc := tt.desc
+		if desc == "" {
+			desc = fmt.Sprintf("%s %s", tt.method, tt.body)
+		}
```

Comment:
All the descriptions are filled in, so I think you can just use `tt.desc` for the t.Run and remove these lines.